### PR TITLE
Removed self assignment of variable in GlobalTrackQualityProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
This fixes a clang warning.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

File: RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc [hunk 1]
```diff
@@ -285,11 +285,9 @@ std::pair<double,double> GlobalTrackQualityProducer::newChi2(Trajectory& muon) c
     }
   }
   
-  if (tkNdof < 6 ) tkChi2 = tkChi2; // or should I set it to  a large number ?
-  else tkChi2 /= (tkNdof-5.);
+  if (tkNdof > 5 ) {tkChi2 /= (tkNdof-5.); }
```
Comment (slava77) [inline RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc]: is it really obvious that ```tkChi2 = tkChi2; // or should I set it to  a large number ?``` was not a typo?

Comment (Dr15Jones) [reply RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc]: From 'blame' that line has been in the code for 10 years.

Comment (Dr15Jones) [reply RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc]: https://github.com/cms-sw/cmssw/blame/master/RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc#L288

Comment (slava77) [reply RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc]: I "refreshed" my memory a bit.
This part was questionable to me back when it was proposed.
At least for historical reason, I propose to leave a comment here about the other values; smth like ```if (tkNdof < 6 ) the tkChi2 is used directly: should a large number or something else be used``` perhaps with a FIXME.

@drkovalskyi is anybody still using these IDs?

**Comment by perrotta** [comment]
@slava77 : I removed my "+1" and I let you follow with it

**Comment by Dr15Jones** [comment]
please test

**Comment by fabiocos** [comment]
+1

**Pull request merged**